app/controllers/users_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.new(user_params)
+    if user.save
+      log_in user
+      remember user
+      redirect_to root_url
+      flash[:success] = 'Account created'
+    else
+      flash[:errors] = user.errors.full_messages
+      render 'new'
+    end
+  end
+
+  def show
```

Comment:
Avoiding repetition of the query to find the user with the params could be achieved by creating a callback that will set the user before running the show method. [OPTIONAL]

```
private

def set_user
  @user = User.find(params[:id])
end
```